charts/kube-prometheus-stack/templates/prometheus/secret.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if .Values.prometheus.enabled }}
+{{- if and .Values.prometheus.enabled .Values.prometheus.prometheusSpec.thanos.objectStorageConfig }}
```

Comment:
What about

.Values.prometheus.prometheusSpec.thanos.objectStorageConfig.existingSecret

?

If defined, then this secret can be omit, too? Like line 13

Reply:
I have applied the suggestion. Moved the if statement to the beginning of the manifest.